backend/app/Main.hs

```diff
@@ -120,19 +120,23 @@ register sessions conn = do
   RegisterParams username (mkPassword -> password) name <- fromBody
   -- TODO: Validation
   hashedPassword <- liftIO $ hashPassword password
+
   _ <- liftIO $ run
     conn
     "INSERT INTO users (username, password, name) VALUES (?, ?, ?)"
     [toSql username, toSql (unPasswordHash hashedPassword), toSql name]
+  
   liftIO $ commit conn
+
   users <- liftIO $
     quickQuery' conn "SELECT id, name FROM users WHERE username = ?" [toSql username]
+
   case users of
     [map fromSql -> [id, name]] -> do
       sessionID <- liftIO nextRandom
       liftIO $ modifyIORef' sessions $ M.insert sessionID id
-      send $ withCookie "SESSION" (toText sessionID) $ json $ LoginResponse name (pack $ show id)
-    _ -> error "impossible"
-  
+      send $ withHeader ("token", toASCIIBytes sessionID) $ withCookie "SESSION" (toText sessionID) $ json $ LoginResponse name (pack $ show id)
+    _ -> send $ status status400 $ text "User already exists"
```

Comment:
This case could either mean that the insertion has failed or that there exists multiple users with the same username. We still need to add validation that no user with that username exists before the insertion anyhow, so this will need to be reworked again in the near future.

Reply:
I agree! Will add a comment before merging.